aws/security_group.tf

```diff
@@ -8,7 +8,7 @@ resource "aws_security_group" "allow_ssh" {
     from_port        = 22
     to_port          = 22
     protocol         = "tcp"
-    cidr_blocks      = ["0.0.0.0/0"]
+    cidr_blocks      = ["<cidr>"]
```

Comment:
**Tenable.cs remediation**
 
**[HIGH]: Ensure no security groups allow ingress from 0.0.0.0/0 to remote server administration ports**
 
**Vulnerability**
Description:

Security groups provide stateful filtering of ingress and egress network traffic to AWS resources. It is recommended that no security group allows unrestricted ingress access to remote server administration ports, such as SSH to port '22' and RDP to port '3389'.

Rationale:

Public access to remote server administration ports, such as 22 and 3389, increases resource attack surface and unnecessarily raises the risk of resource compromise.

When updating an existing environment, ensure that administrators have access to remote server administration ports through another mechanism before removing access by deleting the 0.0.0.0/0 inbound rule.
 
**Remediation**
Perform the following to implement the prescribed state:

1. Login to the AWS Management Console at [https://console.aws.amazon.com/vpc/home](https://console.aws.amazon.com/vpc/home)
2. In the left pane, click 'Security Groups' 
3. For each security group, perform the following:
1. Select the security group
2. Click the 'Inbound Rules' tab
3. Click the 'Edit inbound rules' button
4. Identify the rules to be edited or removed
5. Either A) update the Source field to a range other than 0.0.0.0/0, or, B) Click 'Delete' to remove the offending inbound rule
6. Click 'Save rules'

source: tenablecs-